src/main/java/seedu/bookmark/model/UserPrefs.java

```diff
@@ -15,6 +15,7 @@
 
     private GuiSettings guiSettings = new GuiSettings();
     private Path bookmarkFilePath = Paths.get("data" , "library.json");
+    private String sortingPreference = "";
```

Comment:
You might want to consider using Prefix instead of String. From what I see, you are converting from Prefix to String to set in UserPrefs. In order to get back the Prefix to create a Comparator, you have to convert the String back to a Prefix which seems unnecessary if Prefix was used instead. By using Prefix, you can remove the need for sortingPrefixGenerator and reduce the possibilities of bugs by the back and forth conversions. 
Is there any additional benefit to using a String that I have missed out?